crates/libtest2/src/macros.rs

```diff
@@ -12,10 +12,23 @@ macro_rules! _main_parse {
     };
 }
 
+#[macro_export]
+macro_rules! _parse_ignore {
+    (ignore) => {
+        ::std::option::Option::<&'static str>::None
+    };
+    (ignore = $reason:expr) => {
+        ::std::option::Option::<&'static str>::Some($reason)
+    };
+    ($($attr:tt)*) => {
+        compile_error!(concat!("unknown attribute '", stringify!($($attr)*), "'"));
+    };
+}
+
 #[macro_export]
 #[allow(clippy::crate_in_macro_def)] // accessing item defined by `_main_parse`
 macro_rules! _test_parse {
-    (#[test] fn $name:ident $($item:tt)*) => {
+    (#[test] $(#[$($attr:tt)*])* fn $name:ident $($item:tt)*) => {
```

Comment:
I created #141 for handling `#[cfg]`

I wonder if we should generally allow users attributes through but the question is what do we then put them on.